.github/workflows/vercel-deploy.yml

```
@@ -0,0 +1,131 @@
+      - name: 📦 Install Vercel CLI
+        run: bun add -g vercel@latest
+
+      - name: 🔗 Pull Vercel Environment
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+
+      - name: 🏗️ Build Project
+        run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+
+      - name: 🚀 Deploy Preview
+        id: deploy
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:2:28: Double quote to prevent globbing and word splitting

(shellcheck)